### PR TITLE
[file-system][android] Add support for multipart upload

### LINKED ISSUE #5841: [iOS] Look into implementing Background Upload
Using [NSURLSessionUploadTask](https://developer.apple.com/documentation/foundation/nsurlsessionuploadtask?language=objc)

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Part of #5841.

# How

- js part is here: https://github.com/expo/expo/pull/7403 
- add multipart upload on iOS

# Test Plan

- https://snack.expo.io/@lukaszkosmaty/expo-file-system---add-multipart-upload ✅

### CODE REVIEW

File: packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java [hunk 1]
```diff
@@ -485,14 +485,50 @@ public void uploadAsync(final String fileUriString, final String url, final Map<
...
-      RequestBody body = RequestBody.create(null, uriToFile(fileUri));
-
+      File file = uriToFile(fileUri);
       String method = "POST";
       if (options.containsKey("httpMethod")) {
         method = importHttpMethod((int) (double) options.get("httpMethod"));
       }
 
-      requestBuilder.method(method, body);
+      int type = 0;
```
Comment (bbarthec) [inline packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java]: You know that I cannot allow such dangling numbers 😉

File: packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java [hunk 2]
```diff
@@ -485,14 +485,50 @@ public void uploadAsync(final String fileUriString, final String url, final Map<
...
+      if (options.containsKey("uploadType")) {
+        type = (int) (double) options.get("uploadType");
```
Comment (bbarthec) [inline packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java]: What's this 😄?
Cannot we get is somehow differently?

Comment (lukmccall) [reply packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java]: I don't like it too. However, when we pass `number` from js, it's stored as a `Double`.  Jave doesn't allow conversion like this:
```
Double -> int (where Double is a class). 
```
So, we need to unbox it and then we can safely convert to int.

**Review by bbarthec — APPROVED** [review]
Ok, so LGTM - make sure you've updated CHANGELOG.md though 😉
Ok, I see that this PR is firstly merged into your branch and later on to the `master`, so it might be without CHANGELOG 😉

**Pull request merged**